geo/src/main/java/org/apache/carbondata/geo/QuadTreeCls.java

```diff
@@ -0,0 +1,904 @@
+    if (node.getNodeStatus() == GridData.STATUS_ALL) {
+      Long[] range = node.getGrid().getHashIDRange();
+      result.add(range);
+    } else {
+      // The order of addition is, bottom left, top left, top right, bottom right
+      getSubChildGridRange(node, QuadNode.ChildEnum.BOTTOMLEFT, result);
+      getSubChildGridRange(node, QuadNode.ChildEnum.TOPLEFT, result);
+      getSubChildGridRange(node, QuadNode.ChildEnum.TOPRIGHT, result);
+      getSubChildGridRange(node, QuadNode.ChildEnum.BOTTOMRIGHT, result);
+    }
+  }
+
+  /**
+   * Get the range of a node's child nodes
+   * @param node tree node
+   * @param childType Child node type
+   * @param result  return value
+   */
+  public void getSubChildGridRange(QuadNode node, QuadNode.ChildEnum childType,
+                                   List<Long[]> result) {
+    QuadNode child = node.getChildren(childType);
+    if (child != null) {
+      getNodeGridRange(child, result);
+    }
+  }
+
+  /**
+   * Sorting and merging area results
+   * @param rangeList Region list
+   */
+  public void sortRange(List<Long[]> rangeList) {
+    // When sorting, you only need to sort the first node of long [],
+    // because you can ensure that the interval is not repeated
+    rangeList.sort(new Comparator<Long[]>() {
+      @Override
+      public int compare(Long[] x, Long[] y) {
+        return Long.compare(x[0], y[0]);
+      }
+    });
+  }
+
+  /**
+   * If the ordered data is merged, the number of data segments after merging may be reduced.
+   * @param rangeList Area list, sorted
+   */
+  public void combineRange(List<Long[]> rangeList) {
+    if (rangeList.size() > 1) {
+      for (int i = 0, j = i + 1; i < rangeList.size() - 1; i++, j++) {
+        long previousEnd = rangeList.get(i)[1];
+        long nextStart = rangeList.get(j)[0];
+        if (previousEnd + 1 == nextStart) {
+          rangeList.get(j)[0] = rangeList.get(i)[0];
+          rangeList.remove(i);
+        }
+      }
+    }
+  }
+
+  /**
+   * Get the circumscribed rectangle of a polygon
+   * @param polygon polygon
+   * @return Rectangular area
+   */
+  private QuadRect getOuterRectangle(List<double[]> polygon) {
+    double left = Double.MAX_VALUE;
+    double top = Double.MIN_VALUE;
+    double right = Double.MIN_VALUE;
+    double bottom = Double.MAX_VALUE;
+    for (double[] point : polygon) {
+      if (point[0] < left) {
+        left = point[0];
+      }
+      if (point[0] > right) {
+        right = point[0];
+      }
+      if (point[1] < bottom) {
+        bottom = point[1];
+      }
+      if (point[1] > top) {
+        top = point[1];
+      }
+    }
+    Point2D.Double topLeft = new Point2D.Double(left, top);
+    Point2D.Double bottomRight = new Point2D.Double(right, bottom);
+    return new QuadRect(topLeft, bottomRight);
+  }
+
+  public QuadNode getRoot() {
+    return root;
+  }
+
+  public void clean() {
+    if (root != null) {
+      root.clean();
+      root = null;
+    }
+  }
+}
+
+
```

Comment:
check and remove empty lines in all the places of this PR

Reply:
done